src/Client.php

```diff
@@ -284,158 +169,128 @@ public function captureLastError(array $payload = [])
 
         $exception = new \ErrorException(@$error['message'], 0, @$error['type'], @$error['file'], @$error['line']);
 
-        return $this->captureException($exception, $payload);
+        return $this->captureException($exception, $scope);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getLastEvent()
+    public function getIntegration(string $integrationClassName): ?IntegrationInterface
     {
-        return $this->lastEvent;
+        return $this->integrations[$integrationClassName] ?? null;
     }
 
     /**
-     * {@inheritdoc}
+     * Sends the given event to the Sentry server.
+     *
+     * @param Event $event The event to send
+     *
+     * @return null|string
      */
-    public function getLastEventId()
+    private function send(Event $event): ?string
     {
-        @trigger_error(sprintf('The %s() method is deprecated since version 2.0. Use getLastEvent() instead.', __METHOD__), E_USER_DEPRECATED);
-
-        if (null === $this->lastEvent) {
-            return null;
-        }
-
-        return str_replace('-', '', $this->lastEvent->getId()->toString());
+        return $this->transport->send($event);
     }
 
     /**
-     * {@inheritdoc}
+     * Assembles an event and prepares it to be sent of to Sentry.
+     *
+     * @param array      $payload the payload that will be converted to an Event
+     * @param null|Scope $scope   optional scope which enriches the Event
+     *
+     * @return null|Event returns ready to send Event, however depending on options it can be discarded
      */
-    public function capture(array $payload)
+    private function prepareEvent(array $payload, ?Scope $scope = null): ?Event
     {
-        $event = new Event($this->config);
+        $sampleRate = $this->getOptions()->getSampleRate();
+        if ($sampleRate < 1 && mt_rand(1, 100) / 100.0 > $sampleRate) {
```

Comment:
Isn't this wrong? With a sample rate of 0.3 I want a third of the events, I'm getting two thirds like this. Or am I getting this backwards?

Reply:
It's always weird to wrap my head around this, even though it seems to be so simple.
You are right, if it's 0.3, in two-thirds of the cases this comparison is `true`, which means we return `null` which means we discard it ^^
So only in one-third this evaluation is `false` and we continue to create/send the event.